webpack/scenes/ContentViews/Publish/CVPublishReview.js

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+import { Table, TableHeader, TableBody } from '@patternfly/react-table';
+import { translate as __ } from 'foremanReact/common/I18n';
+import ContentViewIcon from '../components/ContentViewIcon';
+import InactiveText from '../components/InactiveText';
+import ComponentEnvironments from '../Details/ComponentContentViews/ComponentEnvironments';
+import { selectEnvironmentPaths } from '../components/EnvironmentPaths/EnvironmentPathSelectors';
+
+const CVPublishReview = ({
+  details,
+  userCheckedItems,
+}) => {
+  const environmentPathResponse = useSelector(selectEnvironmentPaths);
+  const { results } = environmentPathResponse || {};
+  const environmentsAdded = (environmentResult) => {
+    let environmentsList = [];
+    let library = null;
+    environmentResult.forEach((path) => {
+      const {
+        environments,
+      } = path || {};
+      [library] = environments;
+      const envInPath = environments.filter(env =>
+        userCheckedItems.filter(item => item.id === env.id).includes(env));
+      environmentsList = environmentsList.concat(envInPath);
+    });
+    environmentsList.unshift(library);
+    return environmentsList;
+  };
+
+  const {
+    name, composite, next_version: nextVersion,
+  } = details;
+
+  const columnHeaders = [
+    {
+      title: __('Content view'),
+    },
+    {
+      title: __('Version'),
+    },
+    {
+      title: __('Environments'),
+    },
+  ];
+
+  const rows = [
+    [
+      { title: <><ContentViewIcon composite={composite} description={name} /><InactiveText text={__('Newly published')} /></> },
+      { title: `Version ${nextVersion}` },
+      { title: <ComponentEnvironments environments={environmentsAdded(results)} /> },
+    ],
+  ];
+
+  return (
+    <Table
+      aria-label="Review Table"
+      cells={columnHeaders}
+      rows={rows}
```

Comment:
This is a pretty simple Table; would you consider using `TableComposable` instead?  It's going to allow you to decouple your JSX from the row and column data and make everything look so much cleaner!

Reply:
Updated to use TableComposable 👍🏽